### PR TITLE
AAP-85025 XLAB Clarify report management actions

### PR DESCRIPTION
**Pull request by kcagran**
## Summary
[AAP-85025] Clarify report management actions

This pull request introduces several improvements and updates to the automation dashboard's toolbar actions and the date range filter component. The main focus is on enhancing the handling and validation of custom date ranges, refining the report management actions (including permissions and labels), and ensuring the UI and tests accurately reflect these changes.

**Toolbar Actions and Report Management:**

* The dropdown menu for report actions now uses clearer labels: "Create new report", "Edit report", and "Delete report" instead of the previous "Save report" and "Rename report". The main dropdown is labeled "Report actions" instead of "Save as report". [[1]](diffhunk://#diff-93b1c36dae73e96b23a1678442a7bc9ad4d988b0f2a9eb439a5d22a8bcc27078R88-R118) [[2]](diffhunk://#diff-93b1c36dae73e96b23a1678442a7bc9ad4d988b0f2a9eb439a5d22a8bcc27078L128-R143) [[3]](diffhunk://#diff-8558f7cc46bfea894c32b5ee672739c65d3d473b440daccffc1a89420a05ec28L127-R130) [[4]](diffhunk://#diff-8558f7cc46bfea894c32b5ee672739c65d3d473b440daccffc1a89420a05ec28L202-R205) [[5]](diffhunk://#diff-8558f7cc46bfea894c32b5ee672739c65d3d473b440daccffc1a89420a05ec28L271-R377)
* The "Edit report" action is now always available (regardless of whether the filter state is default or not), but is disabled with a clear message if the user is not a superuser or if required custom date ranges are invalid. [[1]](diffhunk://#diff-93b1c36dae73e96b23a1678442a7bc9ad4d988b0f2a9eb439a5d22a8bcc27078R46-R55) [[2]](diffhunk://#diff-8558f7cc46bfea894c32b5ee672739c65d3d473b440daccffc1a89420a05ec28L214-R254) [[3]](diffhunk://#diff-8558f7cc46bfea894c32b5ee672739c65d3d473b440daccffc1a89420a05ec28L251-R298)
* The logic for enabling/disabling actions is refactored: "Create new report" is disabled for non-superusers or invalid filters, and "Edit report" has its own validation logic. [[1]](diffhunk://#diff-93b1c36dae73e96b23a1678442a7bc9ad4d988b0f2a9eb439a5d22a8bcc27078R46-R55) [[2]](diffhunk://#diff-93b1c36dae73e96b23a1678442a7bc9ad4d988b0f2a9eb439a5d22a8bcc27078R88-R118) [[3]](diffhunk://#diff-8558f7cc46bfea894c32b5ee672739c65d3d473b440daccffc1a89420a05ec28L214-R254)
* All relevant tests are updated to match the new labels, behaviors, and validation logic, including edge cases for permissions and filter validity. [[1]](diffhunk://#diff-8558f7cc46bfea894c32b5ee672739c65d3d473b440daccffc1a89420a05ec28L127-R130) [[2]](diffhunk://#diff-8558f7cc46bfea894c32b5ee672739c65d3d473b440daccffc1a89420a05ec28L202-R205) [[3]](diffhunk://#diff-8558f7cc46bfea894c32b5ee672739c65d3d473b440daccffc1a89420a05ec28L214-R254) [[4]](diffhunk://#diff-8558f7cc46bfea894c32b5ee672739c65d3d473b440daccffc1a89420a05ec28L251-R298) [[5]](diffhunk://#diff-8558f7cc46bfea894c32b5ee672739c65d3d473b440daccffc1a89420a05ec28L271-R377)

**Date Range Filter Improvements:**

* The `ToolbarDateRangeFilter` component now derives the start and end dates directly from `filterValues` (props), ensuring the UI always reflects externally-set values (such as when loading a saved report), not just the values present on mount.
* Local state for the date range is removed in favor of prop-driven state, and setter functions for start/end dates update `filterValues` accordingly.
* Added a clear button for the end date with accessible labeling and test ID for improved testability and accessibility.
* Tests are added and updated to verify correct behavior when filter values are set externally, changed, or cleared, and when the user interacts with the date pickers. [[1]](diffhunk://#diff-5b7f3e63b2fe162a7464713409644cf36b7ea87c3dddb8cac4b15b84ed7e1f92R122-R256) [[2]](diffhunk://#diff-5b7f3e63b2fe162a7464713409644cf36b7ea87c3dddb8cac4b15b84ed7e1f92L2-R2)

These changes improve the reliability, clarity, and accessibility of the report management UI and custom date range handling, while ensuring that permissions and validation are consistently enforced across the application.

## Type of Change

- [x] Bug fix
- [x] Enhancement
- [ ] Tests
- [ ] Documentation
- [ ] Other (please specify)

## Risk Analysis - REQUIRED

- [ ] **High** — Broad platform impact (e.g., SWR config, shared framework components, authentication, routing, API wrappers, build/bundler config). Changes affect multiple workspaces or could cause widespread regressions.
- [x] **Medium** — Scoped but cross-cutting (e.g., shared utility functions, changes to multiple pages within one workspace, component library updates, test infrastructure). Limited blast radius but touches common code paths.
- [ ] **Low** — Narrowly scoped (e.g., single page fix, styling tweak, documentation, test-only changes, copy/string updates). Minimal risk of unintended side effects.

### Ephemeral E2E Tests

Once PR is ready and preliminary checks pass, trigger tests by posting a comment `/run-playwright` on this PR.

> Tests run against a fresh AAP instance (version based on branch).

### Screenshots

image

image

image

image

### CODE REVIEW

**Comment by tiyiprh** [comment]
_Cursor assisted_

Thanks for making the changes - it looks good so far! Sorry to give another round of feedback - I'll update the Jira ticket as well. Two small copy changes:

**1. Rename "Edit report" → "Update report"**

The action saves the current name + filters to the selected report (not name-only). "Update" matches saved-views language, avoids the old "Rename" confusion, and signals earlier that you're persisting what's on screen — "Edit" sounds like opening a form, not necessarily saving.

- `[useAutomationDashboardToolbarActions.tsx](frontend/awx/analytics/automation-dashboard/common/useAutomationDashboardToolbarActions.tsx)` — dropdown label
- `[useCreateEditToolbarFilterSetDialog.tsx](frontend/awx/analytics/automation-dashboard/common/useCreateEditToolbarFilterSetDialog.tsx)` — dialog title (e.g. "Update report")
- Related tests


**2. Update disabled tooltip — use "updating" not "saving"**

When the custom date range is incomplete, the Update tooltip currently says *"Enter a valid custom date range before saving"*. For the Update path, something like *"Enter a valid custom date range before updating"* (Create can keep "saving").

Everything else looks good — default-filter gating fix, modal copy, and validFilters gating on Update all work as expected.

cc @marshmalien

File: frontend/awx/analytics/automation-dashboard/common/useAutomationDashboardToolbarActions.tsx [hunk 1]
```diff
@@ -43,6 +43,16 @@ function getSaveDisabledReason(
   return undefined;
 }
 
+function getEditDisabledReason(
+  superuserDisabledReason: string | undefined,
+  validFilters: boolean,
+  t: (key: string) => string
+): string | undefined {
+  if (superuserDisabledReason) return superuserDisabledReason;
+  if (!validFilters) return t('Enter a valid custom date range before saving');
```
Comment (marshmalien) [inline frontend/awx/analytics/automation-dashboard/common/useAutomationDashboardToolbarActions.tsx]:
```suggestion
  if (!validFilters) return t('Enter a valid custom date range before updating');
```

File: frontend/awx/analytics/automation-dashboard/common/useAutomationDashboardToolbarActions.tsx [hunk 2]
```diff
@@ -75,32 +85,37 @@ export function useAutomationDashboardToolbarActions(props: {
...
                   isDisabled: saveDisabledReason,
                   onClick: () => (filterState ? createToolbarFilterSet(filterState) : {}),
                 },
                 {
                   type: PageActionType.Button,
                   icon: PencilAltIcon,
                   selection: PageActionSelection.None,
-                  label: t('Rename report'),
-                  isDisabled: saveDisabledReason,
+                  label: t('Edit report'),
```
Comment (marshmalien) [inline frontend/awx/analytics/automation-dashboard/common/useAutomationDashboardToolbarActions.tsx]:
```suggestion
                  label: t('Update report'),
```

File: frontend/awx/analytics/automation-dashboard/common/useCreateEditToolbarFilterSetDialog.tsx [hunk 3]
```diff
@@ -106,10 +106,15 @@ export function useCreateEditToolbarFilterSetDialog(
   return useCallback(
     (filterState: IFilterState, filterSet: IDashboardFilterSet) => {
       const newFilterSet = { ...filterSet, filters: JSON.stringify(filterState) };
-      const title = filterSet.id === undefined ? t('Save report') : t('Rename report');
+      const isCreate = filterSet.id === undefined;
+      const title = isCreate ? t('Create new report') : t('Edit report');
```
Comment (marshmalien) [inline frontend/awx/analytics/automation-dashboard/common/useCreateEditToolbarFilterSetDialog.tsx]:
```suggestion
      const title = isCreate ? t('Create new report') : t('Update report');
```

**Comment by marshmalien** [comment]
I agree with @tiyiprh suggestions. @kcagran Do you mind resolving the merge conflicts and rebasing with the latest devel?

**Comment by tiyiprh** [comment]
Thanks for making the changes - it looks good to me!

**Comment by SherinV** [comment]
This will go in after the 8/17 GA - https://redhat.atlassian.net/browse/AAP-85025?focusedCommentId=17804166
It can merge to devel, but backport will have to be after 8/17 GA @marshmalien @kcagran

**Review by keithjgrant — APPROVED** [review]
